packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/__tests__/actions.spec.ts

```diff
@@ -334,6 +336,107 @@ describe('DevWorkspace store, actions', () => {
   });
 
   describe('startWorkspace', () => {
+    describe('updateEditor', () => {
```

Comment:
This test suite is already huge, and I would keep it from growing as much as possible.
My suggestion is to simply mock all necessary functions exported from the `updateEditor.ts` module. So you don't need to mock all the things they call internally.
Obviously, you will need to create another test suite to test `updateEditor.ts` itself.

Reply:
@akurinnoy I think It is the best place for these test suites because they test the start workspace action(methods which are called before start).

Could we leave it as is?

BTW: I have created another test suite to test `updateEditor.ts` itself.